trustfall_core/src/interpreter/hints/vertex_info.rs

```diff
@@ -194,7 +169,24 @@ impl<T: InternalVertexInfo + super::sealed::__Sealed> VertexInfo for T {
                 .collect::<Vec<RequiredProperty>>(),
         );
 
-        properties
+        properties.extend(current_component.vertices.values().flat_map(|v| {
+            v.filters
+                .iter()
+                .filter(|f| match f.right() {
+                    Some(Argument::Tag(FieldRef::ContextField(ctx))) => {
+                        current_vertex.vid == ctx.vertex_id
+                    }
+                    Some(Argument::Tag(FieldRef::FoldSpecificField(fsf))) => {
+                        current_vertex.vid == fsf.fold_root_vid
```

Comment:
This is a good question: should we ever return fold-specific fields like count in `required_property`? (And whichever way we decide, we should have a test case to back it up.)

My gut feeling is "no" because they aren't actually properties on the vertex, but derived values based on edges of the vertex.

What do you think?

Reply:
I agree with you, removing it